core/src/main/resources/error/error-classes.json

```diff
@@ -177,6 +177,11 @@
     ],
     "sqlState" : "22018"
   },
+  "CREATE_PERSISTENT_OBJECT_OVER_TEMP_OBJECT" : {
+    "message" : [
+      "Cannot create a persistent <obj> <objName> by referencing a temporary <tempObj> <tempObjName>. Please make the temporary <tempObj> persistent, or make the persistent <obj> temporary."
```

Comment:
Is there any difference among `persistent` and `permanent`? The original errors talk about `permanent` but your named the object differently. Just wonder why?

Reply:
I think they are used interchangeably in Spark. Any preference here @srielau?